File: backend/src/main/java/com/langleon/dsobuildsim/collectorbagbonus/dto/definition/CollectorBagBonusDefinitionDTO.java
package com.langleon.dsobuildsim.collectorbagbonus.dto.definition;

import com.langleon.dsobuildsim.collectorbagbonus.enums.CollectorBagBonusType;
import com.langleon.dsobuildsim.common.StatType;

import java.util.EnumMap;

public record CollectorBagBonusDefinitionDTO(
        CollectorBagBonusType type,
        String name,
        EnumMap<StatType, Double> stats
) {
}

File: frontend/src/app/models/gamedataDTOs/CollectorBagBonusDefinitionDTO.ts
import { CollectorBagBonusType } from "../../enums/CollectorBagBonusType";
import { StatType } from "../../enums/StatType";

export interface CollectorBagBonusDefinitionDTO {
    bonusType: CollectorBagBonusType,
    name: string,
    stats: Record<StatType, number>;
}